Ember js login signup

yuriydyrenko opened the pull request:
Removed the dead logins such as Facebook, Twitter, Google. Also changed the login and signup POST to output JSON. Account page wasn't converted yet.

yuriydyrenko committed:
Basic setup for login

yuriydyrenko committed:
Basic setup for login

yuriydyrenko committed:
Login HTML

yuriydyrenko committed:
Login HTML

yuriydyrenko committed:
Tell nodemon to ignore changes to public/js

yuriydyrenko committed:
Tell nodemon to ignore changes to public/js

yuriydyrenko committed:
Added simple auth

yuriydyrenko committed:
Added simple auth

yuriydyrenko committed:
Initial rough work

yuriydyrenko committed:
Initial rough work

yuriydyrenko committed:
Renamed and moved the auth controller

yuriydyrenko committed:
Renamed and moved the auth controller

yuriydyrenko committed:
Changed to require()

yuriydyrenko committed:
Changed to require()

yuriydyrenko committed:
Added logout

yuriydyrenko committed:
Added logout

yuriydyrenko committed:
Added logout

yuriydyrenko committed:
Added logout

yuriydyrenko committed:
Added template ifs for login

yuriydyrenko committed:
Added template ifs for login

yuriydyrenko committed:
Signup route/controller/template

yuriydyrenko committed:
Signup route/controller/template

yuriydyrenko committed:
Signup route

yuriydyrenko committed:
Signup route

yuriydyrenko committed:
Login error messages

yuriydyrenko committed:
Login error messages

yuriydyrenko committed:
Fixed getting values from autofill

yuriydyrenko committed:
Fixed getting values from autofill

yuriydyrenko committed:
Cleanup

yuriydyrenko committed:
Cleanup

yuriydyrenko committed:
Signup template

yuriydyrenko committed:
Signup template

yuriydyrenko committed:
Signup errors

yuriydyrenko committed:
Signup errors

yuriydyrenko committed:
Signup controller progress

yuriydyrenko committed:
Signup controller progress

yuriydyrenko committed:
Refactoring authenticator controller

yuriydyrenko committed:
Refactoring authenticator controller

yuriydyrenko committed:
Signup errors

yuriydyrenko committed:
Signup errors

yuriydyrenko committed:
Converted signup API to JSON output

yuriydyrenko committed:
Converted signup API to JSON output

yuriydyrenko committed:
Refactor

yuriydyrenko committed:
Refactor

yuriydyrenko committed:
Cleanup and session variables

yuriydyrenko committed:
Cleanup and session variables

yuriydyrenko committed:
User greeting

yuriydyrenko committed:
User greeting

yuriydyrenko committed:
Remove dead code for unused login methods

yuriydyrenko committed:
Remove dead code for unused login methods

yuriydyrenko committed:
Merge branch 'master' into ember-js-login-signup

Conflicts:
	public/js/application.js

willvoutier commented on controllers/user.js:
```diff
@@ -57,21 +57,27 @@ exports.postLogin = function(req, res, next) {
   passport.authenticate('local', function(err, user, info) {
-    if (err) return next(err);
+    if (err) {
+      console.log('first error');
```
kill these

yuriydyrenko replied:
@williamhc I wasn't sure how to handle next(), should we keep them?

willvoutier commented on controllers/user.js:
```diff
@@ -57,21 +57,27 @@ exports.postLogin = function(req, res, next) {
-      if (err) return next(err);
-      return res.redirect('/');
+      if (err) {
+        console.log('login error');
```
and kill these

willvoutier commented on controllers/user.js:
```diff
@@ -105,13 +110,19 @@ exports.postSignup = function(req, res, next) {
+        errors = [{msg: 'Signup error.'}];
       }
-      return res.redirect('/signup');
+      return res.json(errors);
```
we should return a 500 if it's an error: 

`return res.status(500).json(errors);`

willvoutier replied:
also, add a test for this case.

yuriydyrenko replied:
500 is internal server error, I don't think this applies here. Maybe 400, Bad Request? http://www.restapitutorial.com/httpstatuscodes.html

James-Firth replied:
Looking at that link I think 409 is more applicable, thoughts?

yuriydyrenko replied:
I don't think we should change the response code in this case because it causes jQuery's POST request to fail and it makes the code a bit more ugly.

willvoutier commented on controllers/user.js:
```diff
@@ -105,13 +110,19 @@ exports.postSignup = function(req, res, next) {
-      if (err) return next(err);
-      res.redirect('/');
+      if (err) {
+        console.log('signup error');
```
kill these too

willvoutier commented on public/js/config/routes.js:
```diff
@@ -9,6 +9,7 @@ App.Router.map(function() {
   // end generated routes
 
-
+  this.route('login', {path: '/login'});
```
These are the default paths for these route names, you shouldn't need to specify them explicitly.

yuriydyrenko replied:
If those two get removed, an assertion fails and login and signup does not work: `Assertion failed: The attempt to link-to route 'login.index' failed. The router did not find 'login.index' in its possible routes: 'trips', 'trip', 'edit_trip', 'new_trip', 'index'`

willvoutier commented on public/js/templates/index.hbs:
```diff
@@ -1,2 +1,10 @@
 
+{{#if session.isAuthenticated}}
+  Welcome {{ session.name }}<br>
+  ID: {{ session.id }}<br>
```
Is this the session id or the user id? If it's the user's id then it should be called userId or user_id.

yuriydyrenko replied:
@williamhc changed

willvoutier commented on public/js/templates/application.hbs:
```diff
@@ -14,9 +14,30 @@
+                {{else}}
+                  {{#if session.email}}
+                    {{session.email}}
+                  {{else}}
```
When will this else be needed?

yuriydyrenko replied:
@williamhc I don't think we'll need it but I just put it in there because the original .jade template had it like that

willvoutier commented on app.js:
```diff
@@ -91,18 +91,11 @@ app.get( '/trips/:tripid', tripController.showTrip);
 app.delete( '/trips/:tripid', tripController.deleteTrip);
 
-
 app.get('/account', passportConf.isAuthenticated, userController.getAccount);
```
Which of these are still relevant? Delete the ones that aren't (and their associated controller).

willvoutier commented:
A few things:
1. The session doesn't seem to get populated on initial signup, only on login.
2. I think this needs some testing. We should have some tests that are in ember for this since it is such a core interaction (browser tests). We should also have a couple of tests with Mocha to test the signup/login APIs.
3. `/account` errors out. Don't know if we really need an account page, but it shouldn't error out.
4. Do correct me if I'm wrong, but I don't think there is currently a way of setting getting a username with the current setup. We do need usernames for displaying in the UI later. Presenting a user's email as their username is probably not the right idea. :wink: 

Otherwise, just a few minor things. Commented on lines specifically. :cow: :star:

yuriydyrenko committed:
Added new line

yuriydyrenko committed:
Added new line

yuriydyrenko committed:
Changed id to userID

yuriydyrenko committed:
Changed id to userID

yuriydyrenko committed:
Removed dead account stuff

yuriydyrenko committed:
Removed dead account stuff

yuriydyrenko commented:
@williamhc 
1. I forgot to return the same session values on signup as on login. Fixed.
2. Ok.
3. Removed it for now.
4. Your right, no current way to set username on signup. We don't have a username field anyway, its actually user.profile.name. I can add a field for name on signup if you think we should have it.

yuriydyrenko committed:
Removed console.log()

yuriydyrenko committed:
Removed console.log()

willvoutier commented:
We should have a name field on users/profiles, yes.

James-Firth commented:
If it's easy to add I'd agree we should have a username.

For now the email would work though if we have something else to focus on. (eg: itinerary items)

willvoutier committed:
Merge pull request #13 from caseyvill/refactor-trip-controller

Trip controller refactoring

willvoutier committed:
syntax fix - JS objects need commas

yuriydyrenko committed:
Added travis

yuriydyrenko committed:
Testing the login and signup API

yuriydyrenko committed:
Testing the login and signup API

yuriydyrenko committed:
Added Travis-CI status

yuriydyrenko committed:
Removed dead account code

yuriydyrenko committed:
Removed dead account code

yuriydyrenko committed:
Removed dead paths

yuriydyrenko committed:
Removed dead paths

yuriydyrenko committed:
Merge branch 'ember-js-login-signup' of https://github.com/caseyvill/Project-Barbosa into ember-js-login-signup

Conflicts:
	README.md
	app.js
	public/js/application.js

umcook73 commented:
Looks good!
